src/main/java/com/hivemq/cli/commands/options/SubscribeOptions.java

```diff
@@ -48,37 +55,164 @@ public class SubscribeOptions {
+        }
+        userPropertiesFromLegacyOption = true;
+        userProperties = userPropertiesLegacy;
+    }
+
+    private @Nullable File outputFile = null;
+
+    @SuppressWarnings("unused")
+    @CommandLine.Option(names = {"--output-to-file"},
                         description = "A file to which the received publish messages will be written")
-    private @Nullable File outputFile;
+    private void outputFile(final @NotNull File outputFile) {
+        if (this.outputFile != null) {
+            throw new CommandLine.ParameterException(spec.commandLine(),
+                    "A mix of the output file legacy options \"-of\" or \"--outputToFile\" and the current \"--output-to-file\" is used. Please only use \"--output-to-file\" as the legacy options will be removed in a future version.");
+        }
+        this.outputFile = outputFile;
+    }
+
+    @SuppressWarnings("unused")
+    @Deprecated(since = "4.34.0", forRemoval = true)
+    @CommandLine.Option(names = {"-of", "--outputToFile"},
+                        hidden = true,
+                        description = "Options \"-of\" and \"--outputToFile\" are legacy, please use \"--output-to-file\". They will be removed in a future version")
+    private void outputFileLegacy(final @NotNull File outputFileLegacy) {
+        deprecationWarnings.add(
+                "Options \"-of\" and \"--outputToFile\" are legacy, please use \"--output-to-file\". They will be removed in a future version.");
+
+        if (this.outputFile != null) {
+            throw new CommandLine.ParameterException(spec.commandLine(),
+                    "A mix of the output file legacy options \"-of\" or \"--outputToFile\" and the current \"--output-to-file\" is used. Please only use \"--output-to-file\" as the legacy options will be removed in a future version.");
+        }
+        this.outputFile = outputFileLegacy;
+    }
+
+    private boolean isPayloadEncodeToBase64 = false;
 
     @SuppressWarnings("unused")
-    @CommandLine.Option(names = {"-b64", "--base64"},
+    @CommandLine.Option(names = {"--base64"},
                         description = "Specify the encoding of the received messages as Base64 (default: false)")
-    private boolean base64;
+    private void isMessageBase64Encoded(final boolean base64) {
+        if (isPayloadEncodeToBase64) {
+            throw new CommandLine.ParameterException(spec.commandLine(),
+                    "A mix of the base64 legacy options \"-b64\" and the current \"--base64\" is used. Please only use \"--base64\" as the legacy options will be removed in a future version.");
+        }
+        isPayloadEncodeToBase64 = base64;
+    }
+
+    @SuppressWarnings("unused")
+    @Deprecated(since = "4.34.0", forRemoval = true)
+    @CommandLine.Option(names = {"-b64"},
+                        hidden = true,
+                        description = "Option \"-b64\" is legacy, please use \"--base64\". It will be removed in a future version")
```

Comment:
```suggestion
                        description = "Option \"-b64\" is legacy, please use \"--base64\". Legacy options will be removed in a future version")
```

didn't highlight every occurrence, up to you to decide on the change.